packages/overlays/src/OverlayMixin.js

```diff
@@ -158,6 +158,10 @@ export const OverlayMixin = dedupeMixin(
           super.disconnectedCallback();
         }
 
+        if (!this._overlayCtrl) {
+          return;
+        }
+
         this._overlayDisconnectComplete = new Promise((resolve, reject) => {
```

Comment:
Do we rely on this promise somewhere, where we need it even though there is no overlay controller? E.g. for nested overlay nodes, I believe the disconnectedCallback of the nested node gets called before overlay controller is set, and then we do need to make sure that we do some teardown stuff for the content node etc. right?

Reply:
As long as there's no overlay controller (yet), there will be nothing to teardown, right?
I think the overlay setup starts at updateComplete, so the question would be whether `firstUpdated` still fires after `disconnectedCallback`.

Reply:
Actually there is something to teardown https://github.com/ing-bank/lion/blob/master/packages/overlays/src/OverlayMixin.js#L218

The open/close handlers need to be torn down as well as the content node from the global-overlays container. Which I think is why I made the code as it was before this change, to only guard the `this._overlayCtrl.teardown()` with `this._overlayCtrl`  but not guarding `this._teardownOverlayCtrl()` with `this._overlayCtrl`

Does that make sense?

Reply:
Hmm true :\ not sure then... might need to test it with a nested overlay demo to make sure maybe, you can take the nested dialog test case from lion-dialog to check it with, where it creates the nested node first, and then adds it to the parent node template. Because that's the use case where this went weirdly in the past

Reply:
Yes, I thin you mean this story: https://lion-web-components.netlify.app/?path=/story/overlays-system-overview--nested-overlays (this is where the error originally occurred)

Reply:
Actually it's this use case https://github.com/ing-bank/lion/blob/master/packages/overlays/test-suites/OverlayMixin.suite.js#L232. which is slightly different from the demo you linked and also the previous "it" in that test file, because the nested overlay node is constructed before, and then passed as a node into the parent template. This matters, because the connected/disconnected/parent-setupOverlay lifecycles happen in different orders for these different use cases. For the normal use case, it is parent setup overlay --> child connected. When creating the child node first, it's child connected --> parent setup overlay --> child disconnected --> child reconnected.

Reply:
This works (because tests run). It also shows that it reconstructs (it hits the line after check for overlayCtrl in disconnected)